www/docs/migration/migrate-from-v9-to-v10.mdx

```diff
@@ -157,70 +175,82 @@ If you are migrating from V9->V10, the transition is pretty simple for the 95% u
 
-When you've done this, you can start migrating to the new way of doing things.
-
-```ts title='src/server/routers/_app.ts'
-const appRouter = trpc
-  .router<Context>()
-  /* ... */
-  .interop(); // <--- Add this
-export type AppRouter = typeof appRouter;
+```diff title='src/server/routers/_app.ts'
+  const appRouter = trpc
+    .router<Context>()
+    /* ... */
++ .interop();
+  export type AppRouter = typeof appRouter;
 ```
 
-**2. Create the `t`-object**
+When you've done this, you can start migrating to the new way of doing things.
 
-```ts title='src/server/trpc.ts'
-import superjson from 'superjson';
-import { Context } from './context';
+**2. Create the `t`-object**
 
-export const t = initTRPC<{
-  ctx: Context;
-}>()({
-  // Optional:
-  transformer: superjson,
-  // Optional:
-  errorFormatter({ shape }) {
-    return {
-      ...shape,
-      data: {
-        ...shape.data,
-      },
-    };
-  },
-});
+```diff title='src/server/trpc.ts'
++ import superjson from 'superjson';
++ import { Context } from './context';
++
++ export const t = initTRPC<{
++   ctx: Context;
++ }>()({
++   // Optional:
++   transformer: superjson,
++   // Optional:
++   errorFormatter({ shape }) {
++     return {
++       ...shape,
++       data: {
++         ...shape.data,
++       },
++     };
++   },
++ });
 ```
 
 **3. Create a new `appRouter`**
 
 1. Rename your old `appRouter` to `legacyRouter`
 2. Create a new app router:
 
-```ts
-import { t } from './trpc';
+```diff title="src/server/routers/_app.ts"
+  import * as trpc from "@trpc/server";
++ import { t } from "./trpc";
 
-const legacyRouter = trpc
-  .router()
-  /* [...] */
-  .interop();
+- const appRouter = trpc
++ const legacyRouter = trpc
+    .router()
+    /* [...] */
+    .interop();
 
-export const appRouter = t.mergeRouters(legacyRouter);
++ export const appRouter = t.mergeRouters(legacyRouter);
+  export type AppRouter = typeof appRouter;
 ```
 
 3. See if your app still builds
 4. Create a test router:
 
-```ts
-const greetingRouter = t.router({
-  greeting: t.procedure.query(() => 'world'),
-});
+```diff title="/src/routers/greeting.ts"
++ const greetingRouter = t.router({
++   greeting: t.procedure.query(() => 'world'),
++ });
 ```
 
 5. Merge it in:
 
-```ts
-export const appRouter = t.mergeRouters(legacyRouter, greetingRouter);
+```diff title="src/server/routers/_app.ts"
++ import { greetingRouter } from "./routers/greeting";
+
+  /* [...] */
+
+- export const appRouter = t.mergeRouters(legacyRouter);
++ export const appRouter = t.mergeRouters(legacyRouter, greetingRouter);
 ```
 
+### Limitations of interopability mode
+
+Any procedures defined using `t.procedure` must be called on the client using the [new way of calling procedures](#calling-procedures) and vice-versa for old procedures. This means you have to move over an entire router on both the backend and frontend at once.
```

Comment:
> This means you have to move over an entire router on both the backend and frontend at once.

I don't think this is true. You can have an interop router and a new router at the same time and migrate procedures over.

Reply:
+1

Reply:
Ah yeah. Let me rewrite that.

What I meant what when you move that legacyRouter over to be a `t.router` you need to also move the usage of that router on the frontend